src/cljs/main/broadfcui/page/external_importer.cljs

```diff
@@ -48,6 +56,96 @@
+                                (swap! state assoc :load-error (get-parsed-response false))))})))
+   :-render-error
+   (fn [{:keys [state]}]
+     [:div {}
+      [:div {:style {:margin "0.5rem 0"}} "Error loading WDL:"]
+      [comps/ErrorViewer {:error (:load-error @state)}]])
+   :-render-export
+   (fn [{:keys [props state locals this]}]
+     (let [{:keys [id]} props
+           {:keys [banner server-error wdl]} @state]
+       [:div {}
+        (blocker banner)
+        [:div {:style {:marginBottom "1rem"}}
+         (icons/render-icon {:style {:marginRight "0.5rem" :color (:state-warning style/colors)}}
+           :warning)
+         "Please note: Dockstore cannot guarantee that WDL and Docker that is referenced by this Workflow will not change. We advise you to review the WDL before future runs."]
+        [:div {:style {:display "flex"}}
+         [:div {:style {:marginRight "1rem"}}
+          [ExportDestinationForm {:ref #(swap! locals assoc :form %)
+                                  :initial-name (-> id (string/split #"/") last)}]
+          [buttons/Button {:text "Export" :onClick #(this :-export)}]
+          [:div {:style {:marginTop "1rem"}}
+           [comps/ErrorViewer {:error server-error}]]]
+         [:div {:style {:flex "1 1 auto"}}
+          [WDLViewer {:wdl wdl}]]]]))
+   :-export
+   (fn [{:keys [locals this]}]
+     (let [{:keys [form]} @locals]
+       (when (form :valid?)
+         (this :-resolve-workspace (form :get-field-values)))))
+   :-resolve-workspace
+   (fn [{:keys [state this]} {:keys [workspace] :as form-data}]
+     (swap! state assoc :banner "Resolving workspace...")
+     (let [{:keys [existing-workspace new-workspace]} workspace
+           {:keys [project name description auth-domain]} new-workspace]
+       (if existing-workspace
+         (this :-export-method (ws-common/workspace->id existing-workspace) form-data)
+         (endpoints/call-ajax-orch
+          {:endpoint endpoints/create-workspace
+           :payload {:namespace project
+                     :name name
+                     :attributes (if (string/blank? description) {} {:description description})
+                     :authorizationDomain auth-domain}
+           :headers ajax/content-type=json
+           :on-done (fn [{:keys [success? get-parsed-response]}]
+                      (if success?
+                        (this :-export-method {:namespace project :name name} form-data)
+                        (utils/multi-swap! state (assoc :server-error (get-parsed-response false))
+                                                 (dissoc :banner))))}))))
+   :-export-method
+   (fn [{:keys [props state]} {:keys [namespace] :as workspace-id} {:keys [name]}]
+     (swap! state assoc :banner "Exporting method...")
+     (let [config-id (utils/restructure namespace name)]
+       (endpoints/call-ajax-orch
+        {:endpoint (endpoints/post-workspace-method-config workspace-id)
+         :payload {:namespace namespace
```

Comment:
Just wanted to make sure that we are deliberately re-using the workspace's namespace as the config's namespace within the workspace?

Reply:
Yeah, because method config namespaces aren't really a thing any more but are still required by Agora.